tests/test_class_sh_shared_ptr_copy_move.cpp

```diff
@@ -0,0 +1,76 @@
+#include "pybind11_tests.h"
+
+#include <pybind11/smart_holder.h>
+
+#include <memory>
+#include <string>
+#include <vector>
+
+namespace pybind11_tests {
+namespace {
+
+template <int SerNo>
+struct Foo {
+    std::string mtxt;
+    Foo(const std::string &mtxt_) : mtxt(mtxt_) {}
+    Foo(const Foo &other) = delete;
+    Foo(Foo &&other) = delete;
+    std::string get_text() const {
+        std::string res = "Foo";
+        if (SerNo == 0)
```

Comment:
The appveyor build is (and probably all our 30+ MSVC builds would be) failing:
```
C:\projects\pybind11\tests\test_class_sh_shared_ptr_copy_move.cpp(20): warning C4127: conditional expression is constant [C:\projects\pybind11\tests\pybind11_tests.vcxproj]
C:\projects\pybind11\tests\test_class_sh_shared_ptr_copy_move.cpp(22): warning C4127: conditional expression is constant [C:\projects\pybind11\tests\pybind11_tests.vcxproj]
```

My recommendation is to keep the code here as simple as possible (so it's easier to reason about) and to use this approach here:

https://github.com/pybind/pybind11/blob/29fafcc1b28f27512715f28908fd00a946ba5d67/tests/test_class_sh_basic.cpp#L96

Then set the initial `mtxt` from Python.
BTW in some more recent code I called it `history`, if you like that better.

Reply:
Warning fixed and member renamed to ``history``.